easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb

```
@@ -0,0 +1,174 @@
+        'installopts': '-v',
+        'patches': [
+            'torchvision-0.16.2_ffmpeg-6.0-fix.patch',
+            'torchvision-0.16.2_quantized_tol.patch',
+        ],
+        'source_urls': ['https://github.com/pytorch/vision/archive'],
+        'sources': [{'download_filename': 'v%(version)s.tar.gz', 'filename': '%(name)s-%(version)s.tar.gz'}],
+        'checksums': [
+            {'torchvision-0.16.2.tar.gz': '8c1f2951e98d8ada6e5a468f179af4be9f56d2ebc3ab057af873da61669806d7'},
+            {'torchvision-0.16.2_ffmpeg-6.0-fix.patch':
+             'a49336e7bfa1c950e886852bff37a3ea2146ac7bda87241e3ffb31c5cb869cce'},
+            {'torchvision-0.16.2_quantized_tol.patch':
+             '457cdf8ad6653838c552890bce95dbe30b7573b1643334284f5f4a58f74f6e40'},
+        ],
+        'runtest': (
+            'pytest'
+            ' -m "not xfail"'  # don't run tests that are expected that they might fail
+            ' -k "not test_frame_reading_mem_vs_file"'  # this one hangs
+        ),
+        'testinstall': True,
+    }),
+    ('torchaudio', '2.2.1', {
+        'installopts': '--no-use-pep517 -v',
+        'patches': [
+            'torchaudio-2.2.1_use-external-sox.patch',
+            'torchaudio-2.2.1_transform_test_tol.patch',
+        ],
+        'preinstallopts': ('rm -r third_party/{sox,ffmpeg/multi};'  # runs twice when testinstall
+                           ' USE_CUDA=1 USE_OPENMP=1 USE_FFMPEG=1 FFMPEG_ROOT="$EBROOTFFMPEG"'),
+        'source_urls': ['https://github.com/pytorch/audio/archive'],
+        'sources': [{'download_filename': 'v%(version)s.tar.gz', 'filename': '%(name)s-%(version)s.tar.gz'}],
+        'checksums': [
+            {'torchaudio-2.2.1.tar.gz': '5f2a9cda9369ff348d1d7ca2c0a1f01afc9ebff1a13df9775c22aa5c43f5b0f1'},
+            {'torchaudio-2.2.1_use-external-sox.patch':
+             'fc97b2bdf0ab16b9c4706bf2a31dd72d0ecb09e4af2f9c4a34e18fca8e0d165d'},
+            {'torchaudio-2.2.1_transform_test_tol.patch':
+             '381be068dc1c7dda3ea19220e423356ac93621aa48133a0fe54464069c8799c7'},
+        ],
+        'runtest': (
+            'pytest test/torchaudio_unittest/'
+            ' -k "not TestProcessPoolExecutor"'  # hang maybe related https://github.com/pytorch/audio/issues/1021
+            '" and not FilterGraphWithCudaAccel"'  # requires FFmpeg with CUDA support
+            '" and not kaldi_io_test"'  # requires kaldi_io
+        ),
+        'testinstall': True,
+    }),
+    ('pytorch-ignite', '0.4.13', {
+        'modulename': 'ignite',
+        'source_urls': ['https://github.com/pytorch/ignite/archive'],
+        'sources': [{'download_filename': 'v%(version)s.tar.gz', 'filename': '%(name)s-%(version)s.tar.gz'}],
+        'checksums': ['bfe4b6f1cd96e78c021a65a0c51350cdb89d6ef5a8b9609638666ca95bae51d7'],
+        'runtest': (
+            'pytest'
+            ' -m "not multinode_distributed"'
```

```suggestion
            ' -m "not distributed and not multinode_distributed"'
```
> I can't inspect the specifics here, but it seems like most failures here are due to port being busy. I could just exclude all of tests/ignite/distributed/ if those are known to be a bit problematic for people when they build.

This might work in that case.

---

Or just ` --ignore tests/ignite/distributed`.

---

These I think are from distributed as well:
```
            '" and not test__setup_ddp_vars_from_slurm_env_bad_configs"'  # fails sometimes
            '" and not test__native_dist_model_create_from_backend_bad_config"'  # fails sometimes
```
So could probably removed if covered by a more general rule.